internal/controllers/port/controller.go

```diff
@@ -51,7 +51,11 @@ var (
 	subnetDependency = dependency.NewDeletionGuardDependency[*orcv1alpha1.PortList, *orcv1alpha1.Subnet](
 		"spec.resource.addresses[].subnetRef",
 		func(port *orcv1alpha1.Port) []string {
-			subnets := make([]string, len(port.Spec.Resource.Addresses))
+			var subnets []string
```

Comment:
nit: subnets isn't used until line 58. We could remove this declaration and just update line 58 to use `:=`.